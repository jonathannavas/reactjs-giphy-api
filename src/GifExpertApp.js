import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {


    const [categories, setCategories] = useState(['Dbz','Saint Seiya', 'Death note']);

    // const handleAdd = () => {
    //     setCategories( cats => [...cats, 'Futurama'] );
    // };

    return (
        <>
            <h2>GifExpertApp</h2>
            
            <AddCategory setCategories={ setCategories } />

            <hr/>  

            <ol>
                { 
                    categories.map( category => {
                        return <li key={ category } > { category } </li>;
                    })
                }
            </ol>
        </>
    )
}
