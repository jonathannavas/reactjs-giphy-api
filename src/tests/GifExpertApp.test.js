import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Test on GifExpertApp', () => {

    test('Should make a snapshot', ()=>{
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show the categories list', () => {
        const categories = ['dbz','death note'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    

});