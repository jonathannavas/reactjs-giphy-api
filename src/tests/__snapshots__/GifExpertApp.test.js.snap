// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifExpertApp Should make a snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test on GifExpertApp should show the categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="dbz"
      key="dbz"
    />
    <GifGrid
      category="death note"
      key="death note"
    />
  </ol>
</Fragment>
`;
